src/garage/experiment/local_tf_runner.py

```diff
@@ -319,6 +323,39 @@ def train(self,
             pause_for_plot=pause_for_plot,
             start_epoch=0)
 
+        return self._train()
+
+    def _step_epochs(self):
+        """Magic generator for training.
+
+        This function serves as a generator. It is used to separate
+        services such as snapshotting, sampler control from the actual
+        training loop.
+
+        """
+        self.start_worker()
+
+        self.start_time = time.time()
+        self.step_epochs_args = SimpleNamespace(
+            itr=self.train_args.start_epoch * self.train_args.n_epoch_cycles,
+            path=None)
+
+        for epoch in range(self.train_args.start_epoch,
+                           self.train_args.n_epochs):
+            self.itr_start_time = time.time()
+
+            with logger.prefix('epoch #%d | ' % epoch):
+                yield
```

Comment:
Please yield `epoch` here, since it's useful to the user (and is intuitive from the method name).